File: Leetcode/Python/_13.py
class Solution:
    def romanToInt(self, x: str) -> int:
        d = {
            'I':1,
            'V':5,
            'X':10,
            'L':50,
            'C':100,
            'D':500,
            'M':1000
        }
        ans = 0
        i = 0
        while i < len(x):
            if i < len(x)-1 and d[x[i]] <  d[x[i+1]]:
                ans += d[x[i+1]] - d[x[i]]
                i += 2
            else:
                ans += d[x[i]]
                i += 1
        return ans

File: Leetcode/cpp/13.cpp
class Solution {
public:
    int romanToInt(string s) {
        std::unordered_map<char, std::pair<int, int>> umap = {
            {'I', {0, 1}}, {'V', {1, 5}}, {'X', {2, 10}}, {'L', {3, 50}}, {'C', {4, 100}}, {'D', {5, 500}}, {'M', {6, 1000}}
        };
        int output = 0;
        for (int i = s.length() - 1; i >= 0; --i) {
            if (i < s.length() - 1 && umap[s[i]].first < umap[s[i + 1]].first) {
                output -= umap[s[i]].second;
            } else {
                output += umap[s[i]].second;
            }
        }
        return output;
    }
};
